pages/lessons/fundamentals/wolovim-part-1.mdx

```diff
@@ -0,0 +1,321 @@
+
+The only way for the blockchain to verify that asset was truly sent from one user to another is to use assets native to (i.e., created and governed by) that blockchain. In Ethereum, this native currency is called **ether**, and the Ethereum blockchain contains the only official record of account balances.
+
+### **A new paradigm**
+
+This new decentralized tech stack has spawned new developer tools. Such tools exist in many programming languages, but we’ll be looking through the Python lens. To reiterate: even if Python isn’t your language of choice, it shouldn’t be much trouble to follow along.
+
+Python developers that want to interact with Ethereum are likely to reach for **[web3.py](https://web3py.readthedocs.io/)**. web3.py is a library that greatly simplifies the way you connect to an Ethereum node, then send and receive data from it.
+
+<Callout emoji='💡' size='md' variant='info'>
+**Note:** “Ethereum node” and “Ethereum client” are often used interchangeably. In either case, it refers to the software that a participant in the Ethereum network runs. This software can read block data, receive updates when new blocks are added to the chain, broadcast new transactions, and more.
+</Callout>
+
+Ethereum clients can be configured to be reachable by IPC, HTTP, or Websockets. web3.py refers to these connection options as **providers**. You’ll want to choose one of the three providers to configure so that web3.py knows how to communicate with your node.
+
+![https://cdn-images-1.medium.com/max/1600/1*OrElsXOF45w-AgBuezCqjQ.png](https://cdn-images-1.medium.com/max/1600/1*OrElsXOF45w-AgBuezCqjQ.png)
+
+The Ethereum node and web3.py need to be configured to communicate via the same protocol, e.g., IPC in this diagram.
+
+Once web3.py is properly configured, you can begin to interact with the blockchain. Here’s a couple of web3.py usage examples as a preview of what’s to come:
+
+```python
+# read block data:
+w3.eth.get_block('latest')
+
+# send a transaction:
+w3.eth.send_transaction({'from': ..., 'to': ..., 'value': ...})
+```
+
+<SideDrawer buttonText="Did you get all that?" title="Comprehension Check">
+      <Question question="eth-intro-1/2-checkin/Q1" />
+      <Question question="eth-intro-1/2-checkin/Q2" />
+      <Question question="eth-intro-1/2-checkin/Q3" />
+      <Question question="eth-intro-1/2-checkin/Q4" />
+      <Question question="eth-intro-1/2-checkin/Q5" />
+      <Question question="eth-intro-1/2-checkin/Q6" />
+</SideDrawer> 
+<br/>
+
+### **Installation**
+
+In this walkthrough, we’ll just be working within a Python interpreter. We won't be creating any directories, files, classes or functions.
+
```

Comment:
After a lot of juggling around different approaches to make this lesson easily accessible to a complete beginner, a little guide (possibly in the form of a side drawer) right here before installing iPython on setting up a virtual environment would be of immense benefit and clarity.

Reply:
Hey @wolovim 

I was just testing myself to see how quickly I could do a quick guide, and it's probably full of errors, and then there's the version control issue, but some things along these lines might be really helpful as a side drawer just before the *Installation* section. Would you be able to throw something like this together? It would really make a great lesson accessible to all levels, otherwise it's a real *cufuffle* trying to figure out what goes where:

How a snake charmer would set up a virtual environment for Python. 
Let us see what version of python you have installed already.
Open a terminal and run
`python`
If that gives you no result, that’s actually good as you don’t have python2 or lower installed, now run:
` python3 `
If you have python 3.7 or greater, you’re ready to set up your virtual environment, if not, download the latest version of (python3)[]

Now, let’s create a directory to host your venv. I’m calling mine `venv_eth_part1`
`bash // create project directory:
mkdir venv_eth_part1

// navigate into directory
cd venv_eth_part1`

Create the virtual environment:
`bash // unix/macOS:
python3 -m venv env 			

// windows:
py -m venv env`
           
Now activate the environment:
`bash // unix/macOS:
source env/bin/activate		

// windows:
.\env\Scripts\activate`

Double check what your current working directory is:
`bash // unix/macOS:
which python	
// output should be:
.../env/bin/python			

// windows:
where python
// output should be:
...\env\Scripts\python.exe`

How do you deactivate your virtual environment? Easy :) :
`// unix/macOS:
deactivate					

//windows:
deactivate`

If and when you need to use a virtual environment again with all the same dependencies, navigate back to your directory and simply run:
`bash // unix/macOS:
source env/bin/activate		

// windows:
.\env\Scripts\activate`

Now you are ready to set up your dependencies inside your new safe environment, so head back to Installation 👈🏼